### PR TITLE
[3883] Create site dialog updates

### PR DESCRIPTION
**Pull request by jvega190**
3.1.7 Ticket
https://github.com/craftercms/craftercms/issues/3883

### CODE REVIEW

File: ui/app/src/components/CreateSiteDialog.tsx [hunk 1]
```diff
@@ -60,7 +56,7 @@ import Cookies from 'js-cookie';
 import { backgroundColor } from '../styles/theme';
 // @ts-ignore
 import { fadeIn } from 'react-animations';
-
+import { Checkbox, FormControlLabel } from '@material-ui/core';
```
Comment (rart) [inline ui/app/src/components/CreateSiteDialog.tsx]: Import individually from their own file

File: ui/app/src/components/CreateSiteDialog.tsx [hunk 2]
```diff
@@ -556,6 +563,12 @@ function CreateSiteDialog(props: CreateSiteDialogProps) {
     }
   }
 
+  function handleShowIncompatibleChange(e: ChangeEvent<HTMLInputElement>) {
+    setBlueprints(null);
```
Comment (rart) [inline ui/app/src/components/CreateSiteDialog.tsx]: Why clear blueprints? This only involves the marketplace.

File: ui/app/src/components/CreateSiteDialog.tsx [hunk 3]
```diff
@@ -842,6 +855,17 @@ function CreateSiteDialog(props: CreateSiteDialogProps) {
                   className={clsx(classes.tabIcon, search.searchSelected && 'selected')}
                   onClick={handleSearchClick}
                 />
+                <FormControlLabel
+                  className={classes.showIncompatible}
+                  control={
+                    <Checkbox
+                      checked={site.showIncompatible}
+                      onChange={ (e: ChangeEvent<HTMLInputElement>) => {handleShowIncompatibleChange(e)} }
```
Comment (rart) [inline ui/app/src/components/CreateSiteDialog.tsx]: Format (spaces, remove curly braces from inner function or break line). Also move `/>` to it's own line.

File: ui/app/src/components/CreateSiteDialog.tsx [hunk 4]
```diff
@@ -842,6 +855,17 @@ function CreateSiteDialog(props: CreateSiteDialogProps) {
...
+                      color="primary" />
+                  }
+                  label={<Typography className={classes.showIncompatibleInput}>{formatMessage(messages.showIncompatible)}</Typography>}
```
Comment (rart) [inline ui/app/src/components/CreateSiteDialog.tsx]: Format

File: ui/app/src/components/PluginDetailsView.tsx [hunk 5]
```diff
@@ -238,17 +237,21 @@ export default function PluginDetailsView(props: PluginDetailsViewProps) {
         <Typography variant="h5" component="h1">
           {name}
         </Typography>
-        <Button variant="contained" color="primary" className={classes.useBtn} onClick={() => onBlueprintSelected(blueprint, 1)}>
-          {formatMessage(messages.use)}
-        </Button>
+        {
+          compatible &&
+          <Button variant="contained" color="primary" className={classes.useBtn}
+                  onClick={() => onBlueprintSelected(blueprint, 1)}>
```
Comment (rart) [inline ui/app/src/components/PluginDetailsView.tsx]: Format

File: ui/app/src/services/marketplace.ts [hunk 6]
```diff
@@ -17,7 +17,7 @@
 import { get, post } from "../utils/ajax";
 import { MarketplaceSite } from "../models/Site";
 
-export function fetchBlueprints() {
+export function fetchBlueprints(showIncompatible: boolean) {    //TODO: API still not updated with the param
```
Comment (rart) [inline ui/app/src/services/marketplace.ts]: Make the function receive a `options` object with one of the options being `showIncompatible`. Include also type and limit (i.e. all of the currently hard coded params). Assign defaults to all so it keeps working the same way. Make the argument optional. 

Find out from @jdrossl what the API will look like so you can move forward with using the value.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/CreateSiteDialog.tsx (hunk 2 above)
Comment (jvega190) [reply ui/app/src/components/CreateSiteDialog.tsx]: I made this for both bps and marketplace, changing it to only MP

File: ui/app/src/services/marketplace.ts (hunk 6 above)
Comment (jvega190) [reply ui/app/src/services/marketplace.ts]: @rart API still not defined, talked to Jose Ross and decided to use the `showIncompatible` param. Will add to docs

**Review by rart — APPROVED** [review]
@jvega190 is this ready? Still a draft.

**Comment by jdrossl** [comment]
@rart remember this is a `3.1.7` ticket so it can't be merged until we release `3.1.6`

**Review by rart — APPROVED** [review]

**Pull request merged**